flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -1894,6 +1895,52 @@ object TaskManager {
       override def getValue: Double =
         ManagementFactory.getOperatingSystemMXBean().getSystemLoadAverage()
     })
+    // Preprocessing steps for registering cpuLoad
+    // fetch the method to get process CPU load
+    val getCPULoadMethod: Method = getMethodToFetchCPULoad()
+
+    // Log getProcessCpuLoad method not available for Java 6
+    if(getCPULoadMethod == null){
+      LOG.warn("getProcessCpuLoad method not available in the Operating System Bean" +
+      "implementation for this Java runtime environment\n"+Thread.currentThread().getStackTrace)
+    }
+
+    // define the fetchCPULoad method as per the fetched getCPULoadMethod
+    val fetchCPULoad: (Any) => Double = if (getCPULoadMethod != null) {
+      (obj: Any) => getCPULoadMethod.invoke(obj).asInstanceOf[Double]
+    } else {
+      (obj: Any) => -1
+    }
+
+    metricRegistry.register("cpuLoad", new Gauge[Double] {
+      override def getValue: Double = {
+        try{
```

Comment:
Hey, correct me if I am wrong. I found that Scala needs to initialise a variable during declaration. Hence if the variable is declared in an if else clause, it won't be visible to the rest of the code. And reassignment to the val was giving errors. Hence I have made use of the null check.